### ISSUE TITLE
[FEATURE] Concatenating Transformers

### ISSUE THREAD

**Issue by koaning** [koaning/scikit-lego#305]
In scikit-learn, we sometimes forget when we transform. 

Take PCA. 

```
x -> PCA -> x_pca 
```

We forget the input matrix `X` and instead get a new matrix `X_pca` where we loose information. Many times this might be what you want ... but what if you'd like; 

```
x -> Add(Pca) -> [x, x_pca]
```

It might make sense to add a meta estimator such that we append data instead of replacing it. 

Anybody have strong thoughts about this?

**Comment by koaning** [comment]
I guess a `FeatureUnion` might be a better idea... but then we'll need an `IdentityTransformer` ... unless scikit-learn allready has this.

**Comment by tomderuijter** [comment]
`FeatureUnion` might work functionally, but I thought it duplicates data internally not making it very efficient if you have a lot of transformers in your union.

**Comment by koaning** [comment]
Machine Learning in general isn't efficient ... at all. The question is if it is effective ;) 

Part of me wonders if changing something like `FeatureUnion` is changing something that is just a bit **too** fundamental in scikit-learn. For scikit-lego to work it needs to be a second class citizen. I may be fine with a loss of efficiency from a design perspective. 

In general. 

```
value(clarity) > value(performance)
```

Also ... the reason (probably) that it duplicates the data is because that way you can turn on joblib to process the paths in parallel. Which might actually boost the performance.

**Comment by koaning** [comment]
@MBrouns I think having an `IdentityTransformer` might be good to have, but I am doubting because technically scikit-learn allready supports this via their [FunctionTransformer](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.FunctionTransformer.html#sklearn-preprocessing-functiontransformer).

My gut is suggestiong "screw it, let's just add it ... the whole point of this library is utility" ... wanted to check if you agreed though.

**Comment by MBrouns** [comment]
I'm okay with adding an IdentityTransformer. I'm not sure how often it's useful but considering you can build it on top of the `FunctionTransformer` there's also not really any maintenance overhead

**Comment by koaning** [comment]
IdentityTransformer was added a while ago.
